src/hooks/useCredentialApi.ts

```diff
@@ -150,4 +151,132 @@ const useDeleteCredentialApi = (onAddAlert: (alert: AlertProps) => void) => {
   };
 };
 
-export { useDeleteCredentialApi as default, useDeleteCredentialApi };
+const useAddCredentialApi = (onAddAlert: (alert: AlertProps) => void) => {
+  const { t } = useTranslation();
+
+  const apiCall = useCallback(
+    (payload: CredentialType): Promise<AxiosResponse<CredentialType>> =>
+      axios.post(`${process.env.REACT_APP_CREDENTIALS_SERVICE}`, payload),
+    []
+  );
+
+  const callbackSuccess = useCallback(
+    (response: AxiosResponse<CredentialType>) => {
+      onAddAlert({
+        title: t('toast-notifications.description', {
+          context: 'add-credential',
```

Comment:
optional/future: may have missed this when we did the sources work, but keeping the context strings between `edit` and `add` aligned means we can easily merge the 2 hooks in the future

currently this context uses `add-credential` vs the edit which uses `credential_edit`... changing `add-credential` to be more like `edit` may help, `credential_add`...